src/insights_client/__init__.py

```diff
@@ -95,6 +95,34 @@ def sorted_eggs(eggs):
         return [eggs[1], eggs[0]]
 
 
+def _remove_gpg_home(home):
+    """Clean GPG's temporary home directory at path 'home'.
+
+    :param home: Path to the GPG's temporary home.
+    :type home: str
+    :rtype: None
+    """
+    # Shut down GPG's home agent
+    shutdown_process = subprocess.Popen(
+        ["/usr/bin/gpgconf", "--homedir", home, "--kill", "all"],
+        stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    stdout, stderr = shutdown_process.communicate()
+    if shutdown_process.returncode != 0:
+        log("Could not clean GPG home, got return code {rc}".format(rc=shutdown_process.returncode))
```

Comment:
hm IMHO this message can be easily mislead for the error that happens when executing `shutil.rmtree()`; maybe I'd be more explicit:
```suggestion
        log("Could not kill the GPG agent using gpgconf, got return code {rc}".format(rc=shutdown_process.returncode))
```